build/build.ps1

```diff
@@ -162,23 +162,33 @@
     Write-Host "Compile Umbraco"
     Write-Host "Logging to $log"
 
-    # beware of the weird double \\ at the end of paths
-    # see http://edgylogic.com/blog/powershell-and-external-commands-done-right/
-    &$this.BuildEnv.VisualStudio.MsBuild "$src\Umbraco.Web.UI\Umbraco.Web.UI.csproj" `
-      /p:WarningLevel=0 `
-      /p:Configuration=$buildConfiguration `
-      /p:Platform=AnyCPU `
-      /p:UseWPP_CopyWebApplication=True `
-      /p:PipelineDependsOnBuild=False `
-      /p:OutDir="$($this.BuildTemp)\bin\\" `
-      /p:WebProjectOutputDir="$($this.BuildTemp)\WebApp\\" `
-      /p:Verbosity=minimal `
-      /t:Clean`;Rebuild `
-      /tv:"$($this.BuildEnv.VisualStudio.ToolsVersion)" `
-      /p:UmbracoBuild=True `
+   & dotnet build "$src\Umbraco.Web.UI.NetCore\Umbraco.Web.UI.NetCore.csproj" `
+      --configuration $buildConfiguration `
+      --output "$($this.BuildTemp)\bin\\" `
       > $log
 
-    if (-not $?) { throw "Failed to compile Umbraco.Web.UI." }
+   # get files into WebApp\bin
+   & dotnet publish "$src\Umbraco.Web.UI.NetCore\Umbraco.Web.UI.NetCore.csproj" `
+      --output "$($this.BuildTemp)\WebApp\bin\\" `
+      > $log
+
+    # remove extra files
+    $webAppBin = "$($this.BuildTemp)\WebApp\bin"
+    $excludeDirs = @("$($webAppBin)\Config","$($webAppBin)\refs","$($webAppBin)\runtimes","$($webAppBin)\Umbraco","$($webAppBin)\wwwroot")
+    $excludeFiles = @("$($webAppBin)\appsettings.*","$($webAppBin)\*.deps.json","$($webAppBin)\*.exe","$($webAppBin)\*.config","$($webAppBin)\*.runtimeconfig.json")
+    $this.RemoveDirectory($excludeDirs)
+    $this.RemoveFile($excludeFiles)
```

Comment:
I guess still is almost the same as just doing a `dotnet build` and specific release mode and output folder? Did you try that?

Reply:
I used `dotnet build "$src\Umbraco.Web.UI.NetCore\Umbraco.Web.UI.NetCore.csproj" --configuration Release --output "$($this.BuildTemp)\bin\\"` a few lines above and I thought it would be more appropriate to use a different command than doing a build again with a different output folder. And I can't add 2 folder destinations in the same command.

But I believe I would still need to remove all the unnecessary files - files that were taken care of by MsBuild before 🤷